### ISSUE TITLE
Example of how to use sourceMaps and cleancss

### ISSUE THREAD

**Issue by AlexKey** [plus3network/gulp-less#123]
It would be great to have an example to tie together gulp-less and cleanss to generate map files.

The cleancss docs suggest "If input styles are a product of CSS preprocessor (LESS, SASS) an input source map can be passed as a string."

Gulp-less suggests using cleancss as a plugin, however gulp-sourcemaps for sourceMap creation.

I'm unsure if these play nice together?

**Comment by contra** [comment]
@AlexKey I have no idea how less plugins work with sourcemaps, want to test it out?

Personally I use gulp-uncss or another css minifier but a lot of them don't support sourcemaps

**Comment by AlexKey** [comment]
Thanks @contra, I'm having a good'ol play - if I find anything I'll let you know.

**Comment by AlexKey** [comment]
Instead of using the cleancss less plugin, using the [gulp-minify-css](https://github.com/jonathanepollack/gulp-minify-css) I *think* it should work as they are both listed as [gulp-sourcemap compatible](https://github.com/floridoo/gulp-sourcemaps/wiki/Plugins-with-gulp-sourcemaps-support)

With this in mind, I thought something like this would work, but I'm not having much luck so far.
```javascript
    .pipe(sourcemaps.init())
      .pipe(less())
      .pipe(minifyCSS())
    .pipe(sourcemaps.write())
```

I thought about having two seperate tasks, and using loadMaps: true on the sourcemaps init to tie them together, but also no luck.

**Comment by contra** [comment]
No, that should work if gulp-minify-css is listed as compatible.

**Comment by contra** [comment]
Open an issue on that repo?

**Comment by AlexKey** [comment]
Thanks @contra I've started to write some unit tests in the gulp-minify-css to try and recreate the scenario.

I'll close this ticket for the moment and perhaps re-open it once it's sorted, as examples would be handy.

**Comment by bassjobsen** [comment]
> The cleancss docs suggest "If input styles are a product of CSS preprocessor (LESS, SASS) an 
> source map can be passed as a string."

That's exactly what the [Less Clean Css plugin](https://github.com/less/less-plugin-clean-css/) does. 

When gulp-sourcemaps is enable gulp-less runs less with the `{sourceMapFileInline: true}` option. So less should return the correct sourcemap after running cleancss

**Comment by AlexKey** [comment]
Thanks @bassjobsen does this mean that the sourcemap will be inline in the final minified css? I'm hoping not, as it'll cause the css to be chunky?

**Comment by bassjobsen** [comment]
Yes it is in the final CSS. You should define two task, one for development (sourcemaps no minification) and one for production.

**Comment by AlexKey** [comment]
Thanks @bassjobsen but I'm looking to use souremaps in production to be able to refer back to real source.

But thanks for the help.

Good news is that I was able to stitch everything together if I ran all the tooling by hand, lessc to output sourcemaps, and then cleancss to minify the css with an input source map. Then append the sourcemap url by hand. So in theory it could work, I'll update this once I've got it working.
